File: 347-top-k-frequent-elements/top-k-frequent-elements.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = Counter(nums)

        data = [ [] for _ in range(len(nums) + 1)]
        print(map)

        for key, val in map.items():
            data[val].append(key)
        print(data)

        ans = []

        for i in range(len(data)-1,-1,-1):
            if len(data[i]) > 0:
                for j in range(len(data[i])):
                    ans.append(data[i][j])
                    if len(ans) == k:
                        return ans
            
        

File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> map;
        vector<vector<int>> v(nums.size() +1 , vector<int>());
        for(auto& it: nums){
            map[it]++;

        }

        for(auto it: map){
            v[it.second].push_back(it.first);
        }
        vector<int> ans;

        for(int i = nums.size(); i >=0 ; i--){
            for(auto it:v[i]){
                if(ans.size()==k){
                    break;
                }
                ans.push_back(it);
            }
        }

        return ans;

        
    }
};